sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetricInfo.scala

```diff
@@ -27,4 +27,5 @@ import org.apache.spark.annotation.DeveloperApi
 class SQLMetricInfo(
     val name: String,
     val accumulatorId: Long,
-    val metricType: String)
+    val metricType: String,
+    val aggregateMethod: (Array[Long], Array[Long]) => String)
```

Comment:
The only benefit is to not change existing tests. It is pretty minor to me. Except for test code, these objects for the classes like `SQLMetricInfo` are created from `SQLMetric` and we always need to assign the aggregate method from a `SQLMetric`.  It doesn't make sense to have default aggregate method in these classes.

Reply:
Yes, precisely, `SQLMetricInfo` objects are created from `SQLMetric` objects and `SQLPlanMetric` objects are created from `SQLMetricInfo` objects. Most of the time `SQLMetric` objects just need the default aggregateMethod. Hence it is natural for `SQLMetricInfo` and `SQLPlanMetric` to have this same default.
A default is just that; in the non-test parts that this PR needed to touch, we have to explicitly set the aggregateMethod from the `SQLMetric`, because they have to apply to all `SQLMetric`s. But in other cases, for the test parts now, or for future uses that we don't know about now, the default may be applicable.

Reply:
For now the only reason to have default there, is to avoid changing test code. This doesn't look like strong reason. For personal preference, I can understand you may prefer to set default there. However, for code structure I don't think a `SQLMetricInfo` or `SQLPlanMetric` should know how to aggregate metric by default. If we actually need these classes to initiate metric aggregation in the future, we may change them. I think we should not adding default values just because of an unknown future usage.

Reply:
I understand your point. It is just a suggestion. Existing unit tests do not exercise custom metrics, which is a new feature. In my opinion, being able to add this new support in a way that does not necessitate any code change in tests is actually a plus; it shows that the implementation is well isolated (which is, of course, not always possible). New tests that test the new feature would obviously exercise the new parts of the code. That is just my opinion. At the end of the day, this is a minor point.